WeakAuras/Transmission.lua

```diff
@@ -316,37 +270,318 @@ ChatFrame_AddMessageEventFilter("CHAT_MSG_BN_WHISPER_INFORM", filterFunc)
+        button:Hide()
+      end
+      if import.children and #import.children > 0 then
+        import:SetText(L["Import Group"])
+      else
+        import:SetText(L["Import"])
+      end
+    end
   end
-  if(ItemRefTooltip.WeakAuras_Desc_Box) then
-    ItemRefTooltip.WeakAuras_Desc_Box:Hide();
+  local width = import.Text:GetStringWidth()
+  import:SetWidth(min(90, width + 30))
+end
+
+local keyToButton = setmetatable(
+  {
+    -- key values, which the user should not be able to control updating
+    id = false,
+    uid = false,
+    controlledChildren = false,
+    parent = false,
+    internalVersion = false,
+    regionType = false,
+    url = false,
+    desc = false,
+    -- trivial
+    anchorPoint      = checkButtons[8],
+    xOffset          = checkButtons[8],
+    selfPoint        = checkButtons[8],
+    frameStrata      = checkButtons[8],
+    anchorFrameType  = checkButtons[8],
+    anchorFrameFrame = checkButtons[8],
+    yOffset          = checkButtons[8],
+    -- not super convinced that these are trivial
+    height           = checkButtons[8], 
+    width            = checkButtons[8],
+    fontSize         = checkButtons[8], 
+    -- trigger
+    trigger             = checkButtons[6],
+    untrigger           = checkButtons[6],
+    disjunctive         = checkButtons[6],
+    additional_triggers = checkButtons[6],
+    activeTriggerMode   = checkButtons[6],
+    numTriggers         = checkButtons[6],
+    -- conditions
+    conditions = checkButtons[5],
+    -- load
+    load = checkButtons[4],
+    -- actions
+    actions = checkButtons[3],
+    -- animation
+    animation = checkButtons[2],
+  },
+  {
+    __index = function(self, key) return key and checkButtons[7] end
+  }
+)
+
+buttonAnchor.import:SetScript("OnClick", function(self)
+    if not self.data then return end
+    -- build closure for DynFrame to use
+    HideUIPanel(ItemRefTooltip)
+    WeakAuras.LoadOptions()
+    WeakAuras.SetImporting(true)
+    buttonAnchor.showCode.codes = nil
+    buttonAnchor.thumbnail.currentThumbnail:Hide()
+    buttonAnchor.thumbnail.currentThumbnail = nil
+    -- TODO: find a way to do this better because uh
+    -- it's ugly as hell
+    local optionsFrame = WeakAuras.OptionsFrame()
+    if not(optionsFrame) then
+      WeakAuras.ToggleOptions()
+      optionsFrame = WeakAuras.OptionsFrame()
+    end
+    if not(WeakAuras.IsOptionsOpen()) then
+      WeakAuras.ToggleOptions()
+    end
+    -- TODO: this function is used in at least one other file and needs to be de-duplicated
+    local function findUnusedId(id)
+      local num = 2;
+      while WeakAurasSaved.displays[id] do
+        id = id.." "..num;
+        num = num + 1;
+      end
+      return id
+    end
+    -- upvalues for our closure, and also clear data so that the next import is not corrupted
+    local incomingData, incomingChildren, diffs = self.data, self.children, self.diffs
+    self.data, self.children, self.diffs = nil, nil, nil
```

Comment:
Clearing here should not be needed, and the comment seems to indicate you thought it was necessary here.

If you only clear in import, then not importing won't clear, and thus likely result in the same corrupted state.

Reply:
So, what would you prefer me to do? Add an OnHide handler to ButtonAnchor to clear the data?